### PR TITLE
Clean up MapMatching.java

### PR DESCRIPTION
**Pull request by stefanholder**
Cleaned up MapMatching.java and moved code from [offline_map_matching branch-1.0](https://github.com/bmwcarit/offline-map-matching/tree/branch-1.0) to separate package.

### CODE REVIEW

File: matching-core/src/main/java/com/graphhopper/matching/MapMatching.java [hunk 1]
```diff
@@ -191,122 +196,146 @@ public MatchResult doWork(List<GPXEntry> gpxList) {
...
-                            + likelyReasonStr
-                            + "observation:" + timeStep.observation + ", candidates: " + getSnappedCandidates(timeStep.candidates));
-                }
-                
-                timeStepCounter++;
-                prevTimeStep = timeStep;
+                throw new RuntimeException("Sequence is broken for submitted track at time step "
+                        + timeStepCounter
+                        + " (" + gpxList.size() + " points). "
+                        + likelyReasonStr
```
Comment (karussell) [inline matching-core/src/main/java/com/graphhopper/matching/MapMatching.java]: what is the reason here to break this in multiple lines?

File: matching-core/src/main/java/com/graphhopper/matching/mapmatchingutils/Distributions.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package com.graphhopper.matching;
+package com.graphhopper.matching.mapmatchingutils;
```
Comment (karussell) [inline matching-core/src/main/java/com/graphhopper/matching/mapmatchingutils/Distributions.java]: I know this belongs to the previous PR but just wanted to mention it here: I would prefer just "com.graphhopper.matching.util" which would make it more align with e.g. "com.graphhopper.util" or "com.graphhopper.routing.util"

**Comment by karussell** [comment]
As many changes are also regarding code formatting, would you mind to [check this](https://github.com/graphhopper/graphhopper/issues/770#issuecomment-246490515). Or probably we should apply the new code formatting to the whole project at a later point?

File: matching-core/src/main/java/com/graphhopper/matching/MapMatching.java (hunk 1 above)
Comment (stefanholder) [inline matching-core/src/main/java/com/graphhopper/matching/MapMatching.java]: The line with "throw new RuntimeException(..." was longer than 100 characters. However, I removed line breaks of the following lines to reduce the number of lines.

File: matching-core/src/main/java/com/graphhopper/matching/mapmatchingutils/Distributions.java (hunk 2 above)
Comment (stefanholder) [inline matching-core/src/main/java/com/graphhopper/matching/mapmatchingutils/Distributions.java]: OK, I renamed the package to com.graphhopper.matching.util.

**Comment by stefanholder** [comment]
>As many changes are also regarding code formatting, would you mind to check this. Or probably we should apply the new code formatting to the whole project at a later point?

I didn't realize before that the [Graphhopper code formatting rules](https://github.com/graphhopper/graphhopper/blob/master/.github/CONTRIBUTING.md) were so recent. However, I think that the tabs I accidentally inserted in #66 should definitely be converted to spaces because mixing tabs and spaces is really bad.

If you prefer I can undo the line break changes. However, I only changed MapMatching.java, which should cause no trouble with pending pull requests and you can still apply the new code formatting to the whole project later.

**Comment by karussell** [comment]
Thanks!

> However, I only changed MapMatching.java, which should cause no trouble with pending pull requests and you can still apply the new code formatting to the whole project later.

yes, ok

**Pull request merged**